### PR TITLE
Added new counter to Scheduler: events_fired; small Scheduler cleanup

### LINKED ISSUE #448: Expose scheduler event as a counter
As a DevOps person, I want to monitor the efficiency of the simulators themselves.  Key simulator performance metrics are "Events per Instruction", "kilo-events per second" and "kilo-instructions per second".  "Instructions" are concept of a CPU core performance model, so the simulator is responsible for that metric.  Sparta already has time counters, so all I am missing is "events".

Request: Please add a counter for events, so I can use the Sparta reporting subsystem to write reports on simulator performance.

### PR DESCRIPTION
**Pull request by knute-sifive**
closes #448

### CODE REVIEW

File: sparta/sparta/kernel/Scheduler.hpp [hunk 1]
```diff
@@ -931,36 +931,36 @@ class Scheduler : public RootTreeNode
     std::unique_ptr<DAG> dag_;
 
     //! The number of groups in the DAG after finalization.
-    uint32_t dag_group_count_;
+    uint32_t dag_group_count_ = 1;
 
     //! The number of firing groups (dag_group_count_ [+pre] [+post])
-    uint32_t firing_group_count_;
+    uint32_t firing_group_count_ = {dag_group_count_ + 2};
```
Comment (benoy-sifive) [inline sparta/sparta/kernel/Scheduler.hpp:937]: Magic number...why 2?

File: sparta/src/Scheduler.cpp [hunk 2]
```diff
@@ -108,9 +94,10 @@ Scheduler::Scheduler(const std::string& name, GlobalTreeNode* search_scope) :
...
                        &sset_,
                        "host_wall_time_count_ms/1000.0"),
-    user_time_cnt_  (&sset_, "host_user_time_count_ms",   "User scheduler performance (not simulated time)",   Counter::COUNT_NORMAL, &user_time_),
-    system_time_cnt_(&sset_, "host_system_time_count_ms", "System scheduler performance (not simulated time)", Counter::COUNT_NORMAL, &system_time_),
-    wall_time_cnt_  (&sset_, "host_wall_time_count_ms",   "Wall scheduler performance (not simulated time)",   Counter::COUNT_NORMAL, &wall_time_),
+    events_fired_cnt_(&sset_, "events_fired",              "Scheduler events fired during simulation",          Counter::COUNT_NORMAL, &events_fired_),
+    user_time_cnt_   (&sset_, "host_user_time_count_ms",   "User scheduler performance (not simulated time)",   Counter::COUNT_NORMAL, &user_time_),
+    system_time_cnt_ (&sset_, "host_system_time_count_ms", "System scheduler performance (not simulated time)", Counter::COUNT_NORMAL, &system_time_),
+    wall_time_cnt_   (&sset_, "host_wall_time_count_ms",   "Wall scheduler performance (not simulated time)",   Counter::COUNT_NORMAL, &wall_time_),
```
Comment (benoy-sifive) [inline sparta/src/Scheduler.cpp:100]: In the description, does it make sense to also put "ms" as the units?

**Review by benoy-sifive — APPROVED** [review]

**Review by MichaelSchoenfelder-SiV — APPROVED** [review]
I see `events_fired` as a scheduler stat, so, yes, this is what I am asking for.

File: sparta/sparta/kernel/Scheduler.hpp (hunk 1 above)
Comment (knute-sifive) [reply sparta/sparta/kernel/Scheduler.hpp]: Removed.  This was old code.  The value is actually properly calculated in finalize (and there's a comment there).

**Pull request merged**